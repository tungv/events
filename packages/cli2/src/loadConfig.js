import path from 'path';
import fs from 'fs';

const loadConfig = ({ config, name, redis, port, workers }) => {
  const configValues = tryParseConfig(config);
  const cliOpts = {};

  if (typeof name !== 'undefined') cliOpts.name = name;
  if (typeof redis !== 'undefined') cliOpts.redis = redis;
  if (typeof port !== 'undefined') cliOpts.port = parseInt(port);
  if (typeof workers !== 'undefined') cliOpts.workers = parseInt(workers);

  const fv = Object.assign(configValues, cliOpts);
  return fv;
};

const tryParseConfig = configFilePath => {
  const absPath = path.resolve(process.cwd(), configFilePath);
  try {
    fs.statSync(absPath);
  } catch (ex) {
    // console.warn('file not found', absPath);
    return {};
  }

  try {
    const config = require(absPath);
    return config;
  } catch (ex) {
    console.error(ex);
  }
};

export default loadConfig;
