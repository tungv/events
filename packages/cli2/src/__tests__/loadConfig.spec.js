import loadConfig from '../loadConfig';

describe('helpers: loadConfig', () => {
  it('should load file specified by --config', () => {
    const conf = loadConfig({ config: 'tests/fixtures/heq.config.js' });
    expect(conf).toMatchSnapshot('bare config from file');
  });

  it('should load file and let cli opts override it', () => {
    const conf = loadConfig({
      config: 'tests/fixtures/heq.config.js',
      name: 'override',
      workers: 0,
    });

    expect(conf).toMatchSnapshot('cli options override bare config from file');
  });
});
