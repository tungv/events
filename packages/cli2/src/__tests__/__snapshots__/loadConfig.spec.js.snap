// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers: loadConfig should load file and let cli opts override it: cli options override bare config from file 1`] = `
Object {
  "name": "override",
  "port": 8081,
  "redis": "redis://localhost:6379",
  "workers": 0,
}
`;

exports[`helpers: loadConfig should load file specified by --config: bare config from file 1`] = `
Object {
  "name": "test-1",
  "port": 8081,
  "redis": "redis://localhost:6379",
  "workers": 4,
}
`;
